composer/trainer/checkpoint.py

```diff
@@ -71,22 +72,32 @@ class Checkpointer:
     """Manager for saving state to checkpoint files.
 
     Args:
-        checkpoint_folder (str): The path to the folder to store checkpoints in.
-        checkpoint_interval (int): The amount of time units to wait between checkpoints.
-        checkpoint_interval_unit (str): The unit (`"ep"` or `"it"`) that
-            `checkpoint_interval` should be measured in.
+        checkpoint_folder (str, optional): The folder to save checkpoints to. Relative to the run directory, 
+            (default: ``checkpoints``)
+        checkpoint_interval (int, optional): The frequency with which to checkpoint. (default: ``1``)
+        checkpoint_interval_unit (str, optional): Unit for the checkpoint save interval -- should be 'ep'
+            for epochs, 'it' for iterations, or None to disable checkpointing. (default: ``None``).
     """
 
-    def __init__(self, checkpoint_folder: str, checkpoint_interval: int, checkpoint_interval_unit: str):
-        if checkpoint_interval_unit.lower() == "ep":
-            self.save_event = Event.EPOCH_END
-        elif checkpoint_interval_unit.lower() == "it":
-            self.save_event = Event.BATCH_END
-        else:
-            raise ValueError(f"Unknown checkpointing interval: {checkpoint_interval_unit}")
+    def __init__(self,
+                 checkpoint_folder: str = "checkpoints",
+                 checkpoint_interval: int = 1,
+                 checkpoint_interval_unit: Optional[str] = None):
+        self.checkpoint_interval_unit = checkpoint_interval_unit
         self.checkpoint_folder = checkpoint_folder
         self.save_interval = checkpoint_interval
 
+    @property
+    def _save_event(self):
+        if self.checkpoint_interval_unit is None:
+            raise RuntimeError("Checkpointing is diabled")
+        if self.checkpoint_interval_unit.lower() == "ep":
+            return Event.EPOCH_END
+        elif self.checkpoint_interval_unit.lower() == "it":
+            return Event.BATCH_END
+        else:
+            raise RuntimeError(f"Unknown checkpointing interval: {self.checkpoint_interval_unit}")
```

Comment:
It's a little strange to move this event outside of the init. This seems like it could delay the error by quite a bit when it could have been checked already.

Reply:
This change allows you to change the checkpointing parameters mid-training. Good point though, I'll do a custom setter for checkpoint_interval_unit that performs this validation.